packages/devtools_app/test/memory/control/settings_dialog_test.dart

```diff
@@ -0,0 +1,138 @@
+        .thenAnswer((_) => Future.value(false));
+    setGlobal(ServiceConnectionManager, fakeServiceManager);
+
+    controller.offline.value = true;
+    controller.memoryTimeline.offlineData.clear();
+    controller.memoryTimeline.offlineData.addAll(memoryJson.data);
+  }
+
+  Future<void> pumpMemoryScreen(
+    WidgetTester tester, {
+    MemoryController? memoryController,
+  }) async {
+    await tester.pumpWidget(
+      wrapWithControllers(
+        const MemoryBody(),
+        memory: controller = memoryController ?? MemoryController(),
+      ),
+    );
+
+    // Delay to ensure the memory profiler has collected data.
+    await tester.pumpAndSettle(const Duration(seconds: 1));
+    expect(find.byType(MemoryBody), findsOneWidget);
+  }
+
+  // Set a wide enough screen width that we do not run into overflow.
+  const windowSize = Size(2225.0, 1000.0);
+
+  group('MemoryScreen', () {
+    setUp(() async {
+      await ensureInspectorDependencies();
+      setGlobal(OfflineModeController, OfflineModeController());
+      fakeServiceManager = FakeServiceManager();
+      when(fakeServiceManager.connectedApp!.isDartWebAppNow).thenReturn(false);
+      when(fakeServiceManager.connectedApp!.isDebugFlutterAppNow)
+          .thenReturn(false);
+      when(fakeServiceManager.vm.operatingSystem).thenReturn('android');
+      when(fakeServiceManager.connectedApp!.isDartWebApp)
+          .thenAnswer((_) => Future.value(false));
+      when(fakeServiceManager.errorBadgeManager.errorCountNotifier('memory'))
+          .thenReturn(ValueNotifier<int>(0));
+      setGlobal(ServiceConnectionManager, fakeServiceManager);
+      setGlobal(IdeTheme, IdeTheme());
+      setGlobal(NotificationService, NotificationService());
+      setGlobal(
+        PreferencesController,
+        preferencesController = PreferencesController(),
+      );
+    });
+
+    testWidgetsWithWindowSize('settings update preferences', windowSize,
+        (WidgetTester tester) async {
+      await pumpMemoryScreen(tester);
+      _setUpServiceManagerForMemory();
+
+      // Open the dialog.
+      await tester.tap(find.byType(SettingsOutlinedButton));
+      await tester.pumpAndSettle();
+      await expectLater(
+        find.byType(MemorySettingsDialog),
+        matchesDevToolsGolden('../../goldens/settings_dialog_default.png'),
+      );
+
+      // Modify settings and check the changes are reflected in the controller.
+      expect(preferencesController.memory.autoSnapshotEnabled.value, isFalse);
+      expect(preferencesController.memory.autoSnapshotEnabled.value, isFalse);
+      await tester
+          .tap(find.byKey(MemorySettingDialogKeys.showAndroidChartCheckBox));
+      await tester
+          .tap(find.byKey(MemorySettingDialogKeys.autoSnapshotCheckbox));
+      expect(preferencesController.memory.autoSnapshotEnabled.value, isTrue);
+      expect(preferencesController.memory.autoSnapshotEnabled.value, isTrue);
+      await tester.pumpAndSettle();
```

Comment:
call pumpAndSettle directly after the tap calls instead of after the expect statements